Dockerfile

```
@@ -70,7 +70,10 @@ ARG USER_ID=1000
 # Create a Python virtual environment for use by any application to avoid
 # potential conflicts with Python packages preinstalled in the main Python
 # installation.
-RUN python3.9 -m venv /pipenv-venv
+RUN python3.9 -m venv /pipenv-venv \
+    && /pipenv-venv/bin/python -m pip install --upgrade pip setuptools \
+    && /pipenv-venv/bin/python -m pip install pipenv
+
```

```suggestion
RUN python3.9 -m venv /pipenv-venv \
    && /pipenv-venv/bin/python -m pip install --upgrade pip setuptools
```

The lines below should also be updated:
```diff
- RUN \
-   pip install --upgrade pip && \
-   pip install pipenv
+ RUN pip install pipenv

```

Or if you really want to stick everything into a single layer:
```diff
+ # Install pipenv into the virtual env
- RUN python3.9 -m venv /pipenv-venv
+ RUN python3.9 -m venv /pipenv-venv \
+     && /pipenv-venv/bin/python -m pip install --upgrade pip setuptools \
+     && /pipenv-venv/bin/python -m pip install pipenv

+ ENV PATH="/pipenv-venv/bin:$PATH"
- # Install pipenv into the virtual env
- RUN \
-     pip install --upgrade pip && \
-     pip install pipenv
```

---

I meant that all to be in one layer. I'll remove the additional steps.